### ISSUE TITLE
GCStress=C fails on OS X due to StackOverflowException

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#5761]
**Environment**
OS X x64
COMPlus_GCStress=C

**Sample Output (all tests result in stack overflow)**
```
BEGIN EXECUTION
/Users/russellk/git/coreclr/bin/tests/Windows_NT.x64.Checked/Tests/coreoverlay/corerun badcodeaftercatch_r.exe
Process is terminated due to StackOverflowException.
./badcodeaftercatch_r.sh: line 117: 45913 Abort trap: 6           $_DebuggerFullPath "$CORE_ROOT/corerun" badcodeaftercatch_r.exe $CLRTestExecutionArguments
Expected: 100
Actual: 134
END EXECUTION - FAILED
```

**Comment by RussKeldorph** [comment]
@janvorli @swaroop-sridhar Could one or both of you take a look?

**Comment by RussKeldorph** [comment]
Apparently GCStress=C isn't implemented for OS X yet.

**Comment by RussKeldorph** [comment]
@kouvel in case he can comment.

**Comment by janvorli** [comment]
So, does it mean that the test fails with stack overflow without any GC stress?

**Comment by RussKeldorph** [comment]
@janvorli Tests I tried passed under GCStress=3.  C (presumably 4 and/or 8) were the probem.

**Comment by RussKeldorph** [comment]
@janvorli Tests also pass with no GCStress set.

**Comment by janvorli** [comment]
Oh, I guess I got confused a bit by you saying that  GCStress=C isn't implemented for OS X yet. So what you meant was that we don't support the mode 0xC, but trying to run with it results in the stack overflow, right? So the fix should be to ignore the modes 4 and 8 on OSX for now.

**Comment by RussKeldorph** [comment]
@janvorli It is better to crash and burn than silently ignore, but if you would like to print an informative message that 4 and/or 8 aren't supported and refuse to initialize, that would be fine with me for now.

**Comment by kouvel** [comment]
@adityamandaleeka, it looks like HAVE_GCCOVER used to be defined only on Windows and Linux for CoreCLR and was changed in 0be6c2bb62e3d2528ab37e099d3c387c4237ffc0. I don't have much context into what the build breaks were, could you please take a look?

**Comment by adityamandaleeka** [comment]
@kouvel The build breaks mentioned there were on ARM64 coresys builds.

At that time, it was decided that HAVE_GCCOVER being enabled doesn't mean it actually works on that platform, but just that the code compiles.

**Comment by janvorli** [comment]
I think that having GCStress mode C support is not urgent and we got all the coverage from Linux builds where the GC related stuff is the same as on OSX. So I'd suggest moving it to future.

**Comment by adityamandaleeka** [comment]
@janvorli Yup, I agree. I meant to move it earlier, but was busy with other work. Moved it now.
